apps/rush-lib/src/cli/actions/ListAction.ts

```diff
@@ -25,43 +26,71 @@ export class ListAction extends BaseRushAction {
     this._version = this.defineFlagParameter({
       parameterLongName: '--version',
       parameterShortName: '-v',
-      description: 'If this flag is specified, the project version will be ' +
+      description:
+        'If this flag is specified, the project version will be ' +
         'displayed in a column along with the package name.'
     });
 
     this._path = this.defineFlagParameter({
       parameterLongName: '--path',
       parameterShortName: '-p',
-      description: 'If this flag is specified, the project path will be ' +
+      description:
+        'If this flag is specified, the project path will be ' +
         'displayed in a column along with the package name.'
     });
 
     this._fullPath = this.defineFlagParameter({
       parameterLongName: '--full-path',
       parameterShortName: '-f',
-      description: 'If this flag is specified, the project full path will ' +
+      description:
+        'If this flag is specified, the project full path will ' +
         'be displayed in a column along with the package name.'
     });
+
+    this._jsonFlag = this.defineFlagParameter({
+      parameterLongName: '--json',
+      description: 'If this flag is specified, output will be in JSON format.'
+    });
   }
 
-  protected run(): Promise<void> {
-    return Promise.resolve().then(() => {
+  protected async run(): Promise<void> {
       const allPackages: Map<string, RushConfigurationProject> = this.rushConfiguration.projectsByName;
-      if (this._version.value || this._path.value || this._fullPath.value) {
+      if (this._jsonFlag.value) {
+        this._printJson(allPackages);
+      } else if (this._version.value || this._path.value || this._fullPath.value) {
         this._printListTable(allPackages);
       } else {
         this._printList(allPackages);
       }
+  }
+
+  private _printJson(
+    allPackages: Map<string, RushConfigurationProject>
+  ): void {
+    const objects: string[] = [];
+    allPackages.forEach((_config: RushConfigurationProject, name: string) => {
+     const jsonString: string = JSON.stringify({
```

Comment:
You're still serializing each entry instead of serializing the whole array. This function should look something like

```TypeScript
  private _printJson(allPackages: Map<string, RushConfigurationProject>): void {
    // The IJsonEntry interface will need to be defined
    const projects: IJsonEntry[] = allPackages.map((project: RushConfigurationProject, name: string) => {
      return {
        name,
        version: project.packageJson.version,
        path: project.projectRelativeFolder,
        fullPath: project.projectFolder
      };
    });

    const output: IJsonOutput = { // The IJsonOutput interface will need to be defined
      projects
    };
    console.log(JSON.stringify(output, undefined, 2));
  }
```

Reply:
We also don't want to directly log a rich object because that prints with colors and will screw up piping the result to something that's expecting to parse JSON.